src/backend/Shared/EventBus/EventBus.Extensions.Microsoft.DependencyInjection/ServiceCollectionExtensions.cs

```diff
@@ -3,27 +3,41 @@
 
 using Microsoft.Extensions.DependencyInjection;
 
-using EventBus.Extensions.Microsoft.DependencyInjection.Models;
-using EventBus.Extensions.Microsoft.DependencyInjection.Injection;
+using EventBus.Extensions.Microsoft.DependencyInjection.Settings.Abstract;
+using EventBus.Extensions.Microsoft.DependencyInjection.Configurations.Abstract;
+using EventBus.Extensions.Microsoft.DependencyInjection.ServicesInjectors.Abstract;
+
 using EventBus.Extensions.Microsoft.DependencyInjection.Configurations;
+using EventBus.Extensions.Microsoft.DependencyInjection.ServicesInjectors;
 
 namespace EventBus.Extensions.Microsoft.DependencyInjection
 {
-    public static class ServiceCollectionExtensions
-    {
-        public static IServiceCollection AddRabbitMqEventBus(this IServiceCollection services, Assembly assemblyToScan, Action<RabbitMqConfiguration> configure = null)
-        {
-            RabbitMqConfiguration configuration = new RabbitMqConfiguration();
-            configure?.Invoke(configuration);
-            RabbitMqSettings settings = configuration.BuildSettings();
-
-            ServicesInjector injector = new ServicesInjector();
-            injector.AddRabbitMQPersistentConnection(services, settings);
-            injector.AddSubscriptionManager(services);
-            injector.AddEventBus(services, settings);
-            injector.AddHandlers(services, assemblyToScan);
-
-            return services;
-        }
-    }
+	public static class ServiceCollectionExtensions
+	{
+		public static IServiceCollection AddEventBus<TConfiguration>(this IServiceCollection services, Assembly assemblyToScan, Action<TConfiguration> configure = null)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use the overloading mechanism instead of the optional parameters.](https://app.codacy.com/manual/iamprovidence/Lama/pullRequest?prid=5244170)